crates/fuel-core/src/combined_database.rs

```diff
@@ -419,16 +430,19 @@ impl CombinedDatabase {
                 .latest_height_from_metadata()?
                 .ok_or(anyhow::anyhow!("off-chain database doesn't have height"))?;
 
+            let relayer_db_height = self.relayer().latest_height_from_metadata()?;
+            let relayer_db_rolled_back =
+                is_equal_or_none(relayer_db_height, target_da_height);
+
             let gas_price_chain_height =
                 self.gas_price().latest_height_from_metadata()?;
-
-            let gas_price_rolled_back = gas_price_chain_height.is_none()
-                || gas_price_chain_height.expect("We checked height before")
-                    == target_block_height;
+            let gas_price_rolled_back =
+                is_equal_or_none(gas_price_chain_height, target_block_height);
 
             if on_chain_height == target_block_height
                 && off_chain_height == target_block_height
                 && gas_price_rolled_back
+                && relayer_db_rolled_back
```

Comment:
Relayer database is not related to L2 blocks. So we should revert it independently.

We need a separate function and `fuel-core rollback` should accept another `--da-block-height` argument in order to initiate relayer rollback.

Reply:
Yeah makes sense. Done in https://github.com/FuelLabs/fuel-core/pull/3022/commits/e8c2eeb4b4a388a9815cf8da38362859f965756e